README.md

```diff
@@ -458,6 +458,95 @@ crud.upsert_object('customers',
 ...
 ```
 
+### Upsert many
+
+```lua
+-- Upsert batch of tuples
+local result, err = crud.upsert_many(space_name, tuples, operations, opts)
+-- Upsert batch of objects
+local result, err = crud.upsert_object_many(space_name, objects, operations, opts)
+```
+
+where:
+
+* `space_name` (`string`) - name of the space to insert an object
+* `tuples` / `objects` (`table`) - array of tuples/objects to insert
+* `operations` (`table`) - update [operations](https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_space/#box-space-update) if there is an existing tuple which matches the key fields of tuple  
+* `opts`:
+  * `timeout` (`?number`) - `vshard.call` timeout (in seconds)
+  * `fields` (`?table`) - field names for getting only a subset of fields
+  * `stop_on_error` (`?boolean`) - stop on a first error and report errors
+    regarding the failed operation and all not performed ones, default is
+    `false`
+  * `rollback_on_error` (`?boolean`) - any failed operation will lead to
+    rollback on a storage, where the operation is failed, default is
+    `false`
+
+Returns metadata and array of empty arrays, array of errors
```

Comment:
Why one may need the array of empty arrays? :)

I understand from where it comes:

1. box's `upsert` does not return a tuple in vinyl, because it may skip tuple reading in some cases. It is optimization.
2. crud's `upsert` does not return a tuple for memtx and vinyl both for unification.
3. crud's `upsert`, however, return empty `rows` array. I guess it is also for unification, but with other crud calls.

The third point is quite weak, but since it is just one table it costs almost nothing and why not. Here the situation is different: we'll typically create, encode and transmit over a network 100 extra empty tables per request.

I would just drop the `rows` field.

Reply:
It seems to me that it is better to be resolved in separate PR.

Reply:
I'm okay, but if so, it is better to move the upsert implementation into its own PR completely. I would not release `upsert_many()` with one API and than change it in the backward incompatible way.

Reply:
Seems resolved.